src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -2256,6 +2282,10 @@ public static List<SitePinInst> createMissingSitePinInsts(Design design, Net net
 
                         currPin = net.createPin(pin.getName(), siteInst);
                         newPins.add(currPin);
+                        if (siteInstToNetSiteWiresMap != null) {
+                            siteInstToNetSiteWiresMap.get(siteInst)
```

Comment:
Is it necessary to update this map (here and in a number of places below), since you've already extracted its contents above? Also, `si.getSiteWiresFromNet()` is linear complexity, and returns a whole new list rather than adding to the one that already exists.